packages/math/init.lua

```diff
@@ -67,16 +68,28 @@ function package:registerCommands ()
     xpcall(function()
       mbox = self:ConvertMathML(content, mbox)
     end, function(err) print(err); print(debug.traceback()) end)
-    self:handleMath(mbox, mode)
+    self:handleMath(mbox, mode, nil)
   end)
 
   self:registerCommand("math", function (options, content)
     local mode = (options and options.mode) and options.mode or "text"
     local mbox
+    local counter = (options and options.numbered) and "equation"
+    counter = (options and options.counter) and options.counter or counter
     xpcall(function()
       mbox = self:ConvertMathML(self:compileToMathML({}, self:convertTexlike(content)))
     end, function(err) print(err); print(debug.traceback()) end)
-    self:handleMath(mbox, mode)
+    self:handleMath(mbox, mode, counter)
+  end)
+
+  self:registerCommand("math-counterstyle", function (options, _)
+    SILE.typesetter:typeset("(")
+    SILE.call("show-counter", { id=options.id })
+    SILE.typesetter:typeset(")")
+  end)
+
+  self:registerCommand("equation", function (_, content)
```

Comment:
I might be wrong (untested) but i don't think this will work, due to the fact that this won't trigger the right parser (regular AST vs. raw)

Reply:
Becase you are lucky with "a+b+c", the text is quite innocuous :smile:

Try, and see it failing:
```
\equation{e^{i\pi} = -1}
```

Ouch! When parsing the input `.sil` file, SILE invokes the SIL inputter, which implements the SIL language grammar and builds an abstract syntax tree (AST) for processing. The content of the `\math` command, however is not part of the "usual" language syntax and needs a different grammar. It is, after all, (a subset of) the TeX math syntax, with commands with multiple arguments, special use of brackets, etc.)

The inputter reserves a few special keywords, such as `\script`, `\lua` etc. -- And, you'll now guess it, `\math` too, to tell the parser that the content here should not interpreted as SIL, but just extracted and parsed as a string that will later be parsed by another dedicated inputter (or, in the case of math a pseudo-inputter, not exposed as one, but that's an implementation detail).

Now, we cannot reserve, obviously, too many keywords -- First we have to know them before parsing a file, so it cannot really be dynamic; moreover those can't be overriden or redefined...

Now instead of:
```
self:registerCommand("equation", function (_, content)
    SILE.call("math", {mode="display", counter="equation"}, content)
end)
```

Try this:
```
self:registerRawHandler("equation", function (_, content)
  SILE.call("math", {mode="display", counter="equation"}, content)
end)
```

And in you SIL document, use: `\raw[type=equation]{e^{i\pi} = -1}`
... And it works. As you may also guess, `\raw` is another of the reserved keyworks.

The implication is that SIL cannot have an `\equation` command as you wanted -- Unless the maintainers decide to add it to the list of reserved keywords (_but see below_). It has to go via the `\raw` trick.

That's exactly what a few other packages and inputters do -- With my own markdown.sile, for instance, I use it too, so one can have:
```
\begin[type=markdown]{raw}
Now I am in _[Markdown]{.underline}_\
And $e^{i\pi}$ is a formula.
\end{raw}
```

There's a brief mention of raw handlers in the manual, but it is not really explained. Perhaps something could be added (reusing some of the explanations above is ok, but I'd appreciate if other hands can write a better explanation.

_Unless the maintainers decide to add it to the list of reserved keywords:_ The problem I see there is that we cannot reserve keywords in-definitively for any new use case that pops up. Today `\math`, tomorrow `\equation`, then why not `\formula` later, and so on... And eventually, SIL loses flexible command redefinitions...
Besides, math could have been a 3rd-party package, it does not have to be (strictly speaking) a core feature, or extra ties to the SIL language.
We already have enough reserved keywords, in my opinion. (All things equal, XML has only _one_ similar concept, `<![CDATA[...]]>`.)

But that might be _my_ opinion only.
Anyhow, I hope the above explains the topic well enough.

Reply:
Thank you for the explanation. Some concepts are new to me. In summary, it's not worth creating a (reserved) `\equation` just for the sake of not having to type `\math[...]`. Hence, I removed this command in the revised PR.

Reply:
I opened #1854 to track the documentation aspect. Resolving the conversation here.

Reply:
There are two levels of hackyness I'd like to mention, more for documentation purposes than because it changes anything here.

1. It is possible to pass strings to commands as option values instead of as content, e.g. `\equation[input="e^{i\pi} = -1"]` and fetch it out of `options.input`. This is just moving the cheese though and introduces a whole different set of parsing limitations because of the quoting and other splitting that might go on. I suggest only using this for very special purpose strings like single characters or alternate content where it makes sense to *not* have it marked as content.
2. It might be possible to pass functions some raw text version of their input content as long as we could guarantee that the content *also* parsed correctly into an AST tree. Again this is unlikely to be a good route to go for math syntax, but I could imagine some scenarios where it could be useful for a function to have both access to the parsed AST *and* the text-only pre-parsed version of said text. Note this would not get around our current limitation on raw inputs where one either has to have a balanced number of braces *or* never have the `\end{foo}` sequence corresponding to the tag.